parquet-column/src/main/java/org/apache/parquet/column/statistics/geometry/GeometryStatistics.java

```diff
@@ -0,0 +1,168 @@
+    return coverings;
+  }
+
+  public GeometryTypes getGeometryTypes() {
+    return geometryTypes;
+  }
+
+  public void update(Binary value) {
+    if (value == null) {
+      return;
+    }
+    try {
+      Geometry geom = reader.read(value.getBytes());
+      update(geom);
+    } catch (ParseException e) {
+      abort();
+    }
+  }
+
+  private void update(Geometry geom) {
+    if (supportsBoundingBox()) {
+      boundingBox.update(geom);
+    }
+    if (supportsCovering()) {
+      coverings.stream().forEach(c -> c.update(geom));
+    }
+    geometryTypes.update(geom);
+  }
+
+  /**
+   * A bounding box is a rectangular region defined by two points, the lower left
+   * and upper right corners. It is used to represent the minimum and maximum
+   * coordinates of a geometry. Only planar geometries can have a bounding box.
+   */
+  private boolean supportsBoundingBox() {
+    // Only planar geometries can have a bounding box
+    // based on the current specification
+    return edges == LogicalTypeAnnotation.Edges.PLANAR;
+  }
+
+  /**
+   * A custom WKB-encoded polygon or multi-polygon to represent a covering of
+   * geometries. For example, it may be a bounding box, or an envelope of geometries
+   * when a bounding box cannot be built (e.g. a geometry has spherical edges, or if
+   * an edge of geographic coordinates crosses the antimeridian). In addition, it can
+   * also be used to provide vendor-agnostic coverings like S2 or H3 grids.
+   */
+  private boolean supportsCovering() {
+    // This version (POC) assumes only build coverings for planar edges
+    // In case of spherical edges, no coverings are built
+    //     return edges == LogicalTypeAnnotation.Edges.PLANAR;
+    return true;
+  }
+
+  public void merge(GeometryStatistics other) {
+    Preconditions.checkArgument(other != null, "Cannot merge with null GeometryStatistics");
+    Preconditions.checkArgument(coverings.size() == other.coverings.size(), "Coverings size must be the same");
+
+    boundingBox.merge(other.boundingBox);
+    for (int i = 0; i < coverings.size(); i++) {
+      coverings.get(i).merge(other.coverings.get(i));
+    }
+    geometryTypes.merge(other.geometryTypes);
+  }
+
+  public void reset() {
+    boundingBox.reset();
+    coverings.stream().forEach(c -> c.reset());
+    geometryTypes.reset();
+  }
+
+  public void abort() {
+    boundingBox.abort();
+    coverings.stream().forEach(c -> c.abort());
+    geometryTypes.abort();
+  }
+
+  // Copy the statistics
+  public GeometryStatistics copy() {
+    return new GeometryStatistics(
+        edges,
+        crs,
+        metadata,
+        boundingBox != null ? boundingBox.copy() : null,
+        coverings != null ? coverings : null,
+        geometryTypes != null ? geometryTypes.copy() : null);
+  }
+
+  @Override
+  public String toString() {
+    return "GeometryStatistics{" + "boundingBox="
+        + boundingBox + ", coverings="
+        + coverings + ", geometryTypes="
```

Comment:
The default toString() for `coverings` is not human-readable. We need to print them one by one.